### PR TITLE
Update email regex to accept uppercase & tests

### PR DESCRIPTION
**Pull request by JulienVannier66**
### What does this PR do?

Update email regex to accept uppercase & tests

Related to: https://linear.app/upfluence/issue/ENG-1791/[plg]-cant-register-with-email-that-has-uppercase-letters

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/email-input.ts [hunk 1]
```diff
@@ -10,13 +10,13 @@ interface OSSEmailInputArgs {
   errorMessage?: string;
   validateFormat?: boolean;
   validates?(isPassing: boolean): void;
-  onChange?(value: string | null): void;
+  onChange?(value: string | undefined): void;
```
[the remark this answers is not in the document]
Comment (Miexil) [reply addon/components/o-s-s/email-input.ts:13]: If you're getting a warning on line 50, you'll need to adapt the condition to something like :
`this.args.onChange?.(this.args.value ? this.args.value.toLowerCase : this.args.value);`

File: tests/integration/components/o-s-s/email-input-test.ts [hunk 2]
```diff
@@ -56,11 +56,22 @@ module('Integration | Component | o-s-s/email-input', function (hooks) {
...
-    await render(hbs`<OSS::EmailInput @value={{this.value}} @onChange={{this.onChange}} />`);
-    await typeIn('input', 'a');
-    assert.true(this.onChange.calledOnceWithExactly('a'));
+  module('for the @onChange method', (hooks) => {
+    hooks.beforeEach(function () {
+      this.value = '';
+      this.onChange = sinon.stub();
+    });
+
+    test('it calls after type in new char', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/email-input-test.ts:65]: `@onChange is called when a new character is typed`

File: tests/integration/components/o-s-s/email-input-test.ts [hunk 3]
```diff
@@ -56,11 +56,22 @@ module('Integration | Component | o-s-s/email-input', function (hooks) {
...
+      await render(hbs`<OSS::EmailInput @value={{this.value}} @onChange={{this.onChange}} />`);
+      await typeIn('input', 'a');
+      assert.true(this.onChange.calledOnceWithExactly('a'));
+    });
+
+    test('it sets to lower case the result', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/email-input-test.ts:71]: `if defined, it transforms the result to lowercase`

File: tests/integration/components/o-s-s/email-input-test.ts [hunk 4]
```diff
@@ -56,11 +56,22 @@ module('Integration | Component | o-s-s/email-input', function (hooks) {
...
+      await render(hbs`<OSS::EmailInput @value={{this.value}} @onChange={{this.onChange}} />`);
+      await typeIn('input', 'A');
+      assert.true(this.onChange.calledOnceWithExactly('a'));
```
Comment (Miexil) [inline tests/integration/components/o-s-s/email-input-test.ts:74]: and add a test for `null` input value maybe ?

File: addon/components/o-s-s/email-input.ts [hunk 5]
```diff
@@ -16,7 +16,7 @@ interface OSSEmailInputArgs {
 const DEFAULT_PLACEHOLDER = 'e.g: john.doe@example.com';
 
 export const EMAIL_REGEXP = new RegExp(
-  /^(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])$/
+  /^(?:[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?\.)+[a-zA-Z0-9](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-zA-Z0-9-]*[a-zA-Z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])$/
```
Comment (Miexil) [inline addon/components/o-s-s/email-input.ts:19]: Open question, because I don't know, this was the one defined in creators-web :
`/^\w+([\+\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,})+$/;`

Why is this one (oss / long one) better than the one from creators ? 
@phndiaye @aprentout @OwenCoogan wdyt ?

Comment (phndiaye) [reply addon/components/o-s-s/email-input.ts:19]: I don't remember, to be honest

**Review by phndiaye — APPROVED** [review]

File: addon/components/o-s-s/email-input.ts (hunk 5 above)
Comment (aprentout) [reply addon/components/o-s-s/email-input.ts:19]: The one from creators-web isn't too restrictive ? 
`test++@email.com` which seems to be a valid email is not catch by the one defined in creators-web , I see lot other special case like so during the analysis

**Review by aprentout — APPROVED** [review]

**Review by OwenCoogan — APPROVED** [review]
